src/components/themed/ControlPanel/ControlPanel.js

```diff
@@ -0,0 +1,273 @@
+
+  // ---- Redux State ----
+
+  const activeUsername = useSelector(state => state.core.account.username)
+  const context = useSelector(state => state.core.context)
+
+  /// ---- Local State ----
+
+  // Maintain the list of usernames:
+  const [usernames, setUsernames] = useState(() => arrangeUsers(context.localUsers, activeUsername))
+  useEffect(() => context.watch('localUsers', localUsers => setUsernames(arrangeUsers(context.localUsers, activeUsername))), [context, activeUsername])
+
+  // User List dropdown/open state:
+  const [isDropped, setIsDropped] = useState(false)
+  const handleToggleDropdown = () => {
+    setIsDropped(!isDropped)
+  }
+  useEffect(() => {
+    if (!isDrawerOpen) setIsDropped(false)
+  }, [isDrawerOpen])
+
+  /// ---- Callbacks ----
+
+  const handleDeleteAccount = (username: string) => {
+    Airship.show(bridge => (
+      <ButtonsModal
+        bridge={bridge}
+        title={s.strings.delete_account_header}
+        message={sprintf(s.strings.delete_username_account, username)}
+        buttons={{
+          ok: {
+            label: s.strings.string_delete,
+            async onPress() {
+              await dispatch(deleteLocalAccount(username))
+              return true
+            }
+          },
+          cancel: { label: s.strings.string_cancel }
+        }}
+      />
+    ))
+  }
+
+  const handleSwitchAccount = (username: string) => {
+    dispatch(logoutRequest(username))
+  }
+
+  /// ---- Animation ----
+
+  // Track the destination height of the dropdown
+  const userListMaxHeight = styles.dropdownRow.height * usernames.length + theme.rem(1)
+
+  // Height value above can change if users are added/removed
+  const sMaxHeight = useSharedValue(userListMaxHeight)
+  useEffect(() => {
+    sMaxHeight.value = withTiming(userListMaxHeight)
+  }, [sMaxHeight, userListMaxHeight])
+
+  // Animation completion ratio/multiplier
+  // Shared to sync fade & drop animations
+  const sAnimationMult = useSharedValue(0)
+  useEffect(() => {
+    sAnimationMult.value = withTiming(isDropped ? 1 : 0, {
+      duration: 500,
+      easing: Easing.inOut(Easing.circle)
+    })
+  }, [sAnimationMult, isDropped])
+
+  /// ---- Dynamic CSS ----
+
+  const aDropdown = useAnimatedStyle(() => ({
+    height: sMaxHeight.value * sAnimationMult.value
+  }))
+  const aFade = useAnimatedStyle(() => ({
+    opacity: 0.8 * sAnimationMult.value
+  }))
+  const aRotate = useAnimatedStyle(() => ({
+    transform: [{ rotateZ: `${(isDropped ? -180 : 180) * sAnimationMult.value}deg` }]
+  }))
+
+  return (
+    <SceneWrapper hasHeader={false} hasTabs={false} isGapTop={false} background="none">
+      <View style={styles.panel}>
+        <View style={styles.header}>
+          <View style={styles.logo}>
+            <Image style={styles.logoImage} source={edgeLogo} resizeMode="contain" />
+          </View>
+          <ControlPanelRateComponent />
+          <View>
+            <Pressable onPress={handleToggleDropdown}>
```

Comment:
I think this row needs 2 views.
[icon & userName, left justified][chevron, right justified]